### PR TITLE
[go-gen][metrics] Populate metric.go

### PR DESCRIPTION
**Pull request by jaylees14**
Populate `metric.go`, in accordance with latest `spec-golang`. Haven't dealt with structs yet, since this was getting fairly big anyway.

Updated unit, e2e & integration test

### CODE REVIEW

File: src/main/scala/temple/generate/server/go/service/GoServiceMetricGenerator.scala [hunk 1]
```diff
@@ -1,3 +1,101 @@
 package temple.generate.server.go.service
 
-class GoServiceMetricGenerator {}
+import temple.generate.CRUD.CRUD
+import temple.generate.server.ServiceRoot
+import temple.generate.server.go.common.GoCommonGenerator
```
Comment (willjvsmith) [inline src/main/scala/temple/generate/server/go/service/GoServiceMetricGenerator.scala]:
```suggestion
import temple.generate.server.go.common.GoCommonGenerator._
```

To reduce verbosity

File: src/main/scala/temple/generate/server/go/service/GoServiceMetricGenerator.scala [hunk 2]
```diff
@@ -1,3 +1,101 @@
...
+import temple.generate.utils.CodeTerm.{CodeWrap, mkCode}
+import temple.utils.StringUtils.doubleQuote
+import temple.generate.server.go.common.GoCommonGenerator.genAssign
```
Comment (willjvsmith) [inline src/main/scala/temple/generate/server/go/service/GoServiceMetricGenerator.scala]: And then optimise imports

**Review by willjvsmith — APPROVED** [review]
LGTM 🚀

**Review by Lewiky — APPROVED** [review]
LGTM

**Pull request merged**